react/src/pages/MaintenancePage.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { Button, Card, Row, Col, Typography, theme } from "antd";
+import { RedoOutlined } from "@ant-design/icons";
+import { useTranslation, Trans } from "react-i18next";
+import { useWebComponentInfo } from "../components/DefaultProviders";
+import Flex from "../components/Flex";
+
+const MaintenancePage: React.FC = () => {
+  const { t } = useTranslation();
+
+  const { value, dispatchEvent } = useWebComponentInfo();
+  let parsedValue: {
+    recalculating: boolean;
+    scanning: boolean;
+  };
+  try {
+    parsedValue = JSON.parse(value || "");
+  } catch (error) {
+    parsedValue = {
+      recalculating: false,
+      scanning: false,
+    };
+  }
+  const { recalculating, scanning } = parsedValue;
```

Comment:
How about checking with `typeof`?
```ts
type MaintenanceValue = {
     recalculating: boolean;
    scanning: boolean;
}

const MaintenancePage: React.FC = () => {
  const { t } = useTranslation();

  const { value, dispatchEvent } = useWebComponentInfo();
  const { recalculating, scanning } = typeof value === MaintenanceValue ? JSON.parse(value) : {  recalculating: false, scanning: false }
  
 // ...
 // ...
```

Reply:
oops... I didn't realize the JSON.parse() has to be checked an error and set the default value.
Sorry!

Reply:
But I hope to suggest you to separate the parsing process from the component, like `function parsingValue(value)`.
This can be helpful when someone understands the meaning!